.github/workflows/check-pr.yaml

```diff
@@ -0,0 +1,47 @@
+name: check-pr
+
+on:
+  pull_request:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: true
+
+jobs:
+  check:
+    runs-on: ubuntu-24.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Extract dart version
+      uses: blendfactory/dvm-config-action@1efe1a203d0a3a31bffd767b0affb251b54ff6bc # v2.0.0
+      id: dvm-config-action
+
+      - name: Set up dart
+        uses: dart-lang/setup-dart@e51d8e571e22473a2ddebf0ef8a2123f0ab2c02c # v1.7.1
+        with:
+          sdk: ${{ steps.dvm-config-action.outputs.dart-sdk-version }}
+
+      - name: Install dependencies
+        run: dart pub get
+        shell: bash
+
+      - name: Analyze
+        run: dart analyze
+
+      - name: Check formatting
+        run: dart format --set-exit-if-changed .
+
+     - name: Run test
+        run: dart test --file-reporter="json:test-results.json"
+        continue-on-error: true
+
+      # https://github.com/dorny/test-reporter
+      - name: Report test
+        uses: dorny/test-reporter@31a54ee7ebcacc03a09ea97a7e5465a47b84aea5 # v1.9.1
+        with:
+          name: "Dart Test"
+          path: "test-results.json"
+          reporter: "dart-json"
+          fail-on-error: "true"
```

Comment:
[memo]
I have added the following information as a reference

https://github.com/yumemi-inc/flutter-mobile-project-template/blob/bf2aab61e9519cb20b5bd15663bd16107d4bed51/.github/workflows/check-pr.yaml#L105-L114